src/golang/cmd/server/handler/v2/node_check_results_get.go

```diff
@@ -0,0 +1,147 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aqueducthq/aqueduct/cmd/server/handler"
+	"github.com/aqueducthq/aqueduct/lib/database"
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/aqueducthq/aqueduct/lib/models/shared"
+	"github.com/aqueducthq/aqueduct/lib/models/views"
+	"github.com/aqueducthq/aqueduct/lib/repos"
+	"github.com/aqueducthq/aqueduct/lib/response"
+	"github.com/aqueducthq/aqueduct/lib/storage"
+	"github.com/dropbox/godropbox/errors"
+	"github.com/google/uuid"
+)
+
+// This file should map directly to
+// src/ui/common/src/handlers/v2/NodeCheckResultsGet.tsx
+//
+// Returns all downstream artifact results
+// Route: /api/v2/workflow/{workflowID}/dag/{dagID}/node/check/{nodeID}/results
+// Method: GET
+// Params:
+//	`workflowID`: ID for `workflow` object
+//  `dagID`: ID for `workflow_dag` object
+//	`nodeID`: ID for operator object
+// Request:
+//	Headers:
+//		`api-key`: user's API Key
+// Response:
+//	Body:
+//		`[]response.OperatorWithArtifactResultNode`
+
+type NodeCheckResultsGetHandler struct {
+	nodeGetHandler
+	handler.GetHandler
+
+	Database database.Database
+
+	WorkflowRepo       repos.Workflow
+	DAGRepo            repos.DAG
+	OperatorRepo       repos.Operator
+	OperatorResultRepo repos.OperatorResult
+	ArtifactRepo       repos.Artifact
+	ArtifactResultRepo repos.ArtifactResult
+}
+
+func (*NodeCheckResultsGetHandler) Name() string {
+	return "NodeCheckResultsGet"
+}
+
+func (h *NodeCheckResultsGetHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	return h.nodeGetHandler.Prepare(r)
+}
+
+func (h *NodeCheckResultsGetHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*nodeGetArgs)
+
+	artfID := args.nodeID
+	wfID := args.workflowID
+
+	emptyResponse := []response.OperatorWithArtifactResultNode{}
+
+	dbOperatorWithArtifactNodes, err := h.OperatorRepo.GetOperatorWithArtifactByArtifactIdNodeBatch(ctx, []uuid.UUID{artfID}, h.Database)
+	if err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error reading check node.")
+	}
+	dbOperatorWithArtifactNode := views.OperatorWithArtifactNode{}
+	if len(dbOperatorWithArtifactNodes) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	} else {
+		dbOperatorWithArtifactNode = dbOperatorWithArtifactNodes[0]
+	}
+
+	results, err := h.OperatorResultRepo.GetOperatorWithArtifactResultNodesByOperatorNameAndWorkflow(ctx, dbOperatorWithArtifactNode.Name, wfID, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve check results.")
+	}
+
+	if len(results) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	}
+
+	resultArtifactIds := make([]uuid.UUID, 0, len(results))
+	for _, result := range results {
+		resultArtifactIds = append(resultArtifactIds, result.ArtifactResultID)
```

Comment:
I'd consider making a helper in `functional/slices` called something like `MapWithKey` similar to https://docs.hhvm.com/hsl/reference/function/HH.Lib.Vec.map_with_key/